### ISSUE TITLE
SLPVectorizer.cpp fails to build with gcc 9 on `error: declaration of 'llvm::TargetTransformInfo* llvm::slpvectorizer::BoUpSLP::TTI' changes meaning of 'TTI'`

### ISSUE THREAD

**Issue by sylvestre** [llvm/llvm-project#84049]
```

/build/source/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:3617:24: error: declaration of 'llvm::TargetTransformInfo* llvm::slpvectorizer::BoUpSLP::TTI' changes meaning of 'TTI' [-fpermissive]
 3617 |   TargetTransformInfo *TTI;
      |                        ^~~
In file included from /build/source/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:48:
/build/source/llvm/include/llvm/Analysis/TargetTransformInfo.h:206:29: note: 'TTI' declared here as 'typedef class llvm::TargetTransformInfo llvm::TTI'
  206 | typedef TargetTransformInfo TTI;
      |                             ^~~
```

log: https://llvm-jenkins.debian.net/job/llvm-toolchain-focal-binaries/architecture=amd64,distribution=focal,label=amd64/846/display/redirect

**Comment by patrick-rivos** [comment]
Also seeing this with GCC 11.4. This is a recent regression since I built 57592e9ae78e16ffe4f2a2cd3190c422bcdba5a0 successfully yesterday.

**Comment by patrick-rivos** [comment]
Fixed for me - 083d8aa03aca55b88098a91e41e41a8e321a5721 successfully builds

**Comment by alexey-bataev** [comment]
This is fixed
